crypto/crypto.go

```diff
@@ -42,6 +44,24 @@ func (f SignatureAlgorithm) String() string {
 	return [...]string{"UNKNOWN", "BLS_BLS12381", "ECDSA_P256", "ECDSA_secp256k1"}[f]
 }
 
+// MinSeedLength is a generic minimum seed length to guarantee a minimum entropy.
+// It is used when the seed source is not necessary a CSPRG and the seed
+// should be expanded before being passed to the key generation process.
+const MinSeedLength = crypto.MinSeedLen
+
+// minSeedLength returns the minimum seed length that can safely be used to generate private keys with this algorithm.
+func (f SignatureAlgorithm) minSeedLength() int {
+	switch f {
+	case ECDSA_P256:
+		return crypto.KeyGenSeedMinLenECDSAP256
+	case ECDSA_secp256k1:
+		return crypto.KeyGenSeedMinLenECDSASecp256k1
+	}
+
+	// return a high seed length
+	return 1<<31 - 1
```

Comment:
Hmm, I like the idea to return a high number, but this might cause users to inadvertently allocate 4GB of memory 😬

Reply:
lol you're right, I have made that change at the beginning when the seed length was compared to the output of this function. (although the user allocation will be saved by this 😄  https://github.com/onflow/flow-go-sdk/pull/12/files#diff-279ba67e7dec0c2c6d76346444e8d179R53 )